### DISCUSSION TITLE
Rename an existing instance

### DISCUSSION THREAD

**Discussion by maxenglander** [lima-vm/lima#949] [Q&A]
I would like to rename an existing instance, e.g. `limactl instance rename new-name`. I saw [a similar question](https://github.com/lima-vm/lima/discussions/879), but it didn't seem to be asking the same thing, or the solution didn't seem like what I am looking for.

Is this possible? I did try simply moving the instance folder with `mv ~/.lima/<old-name> ~/.lima/<new-name>` The `<new-name>` showed up in `limactl list`, but when I tried to start the instance after that, it failed.

**Comment by ypresto** [comment]
If you have below error in `/Users/username/.lima/default/ha.stderr.log`, then it is because the **diffdisk** is pointing old path of **basedisk**.

`{"level":"debug","msg":"qemu[stderr]: qemu-system-aarch64: -drive file=/Users/<username>/.lima/<new-name>/diffdisk,if=virtio,discard=on: Could not open backing file: Could not open '/Users/<username>/.lima/<old-name>/basedisk': No such file or directory","time":"2023-06-02T02:53:38+09:00"}`

Then you can fix it by **qemu-img** command (before that, creating backup (`cp diffdisk diffdisk_bak`) is recommended)
```
cp ~/.lima/<new-name>
qemu-img rebase -u -F qcow2 -b basedisk diffdisk
```

**Reply by julienanne** [reply]
Thanks a lot work for me :+1: after my `limactl rename old-name new-name` command failed to start the new instance.
Some informations : `cd` instead of `cp` on last command block and the `cp` command to backup `diffdisk` need to be done after the `cd` command and before the `qemu-img` command

I am totally new to lima but got the same bug so not fixed ? (limactl version b2acc1f)

**Reply by AkihiroSuda** [reply]
> bug

Can you try v2.1.0-rc.0 ? 
`basedisk` and `diffdisk` no longer exist (for newly created instances)
